server/src/storage/datastore.ts

```diff
@@ -38,18 +40,23 @@ const extractAndAppendId = (res: Entity) => {
  * @param kind The Kind that is being queried
  * @param id The id of the Entity being queried
  */
-const getWithId = async (kind: string, id: string) => {
-  const key = datastore.key([kind, datastore.int(id)]);
+const getWithId = async (kind: string, id: number) => {
+  const key = datastore.key([kind, id]);
   const [res] = await datastore.get(key);
   return extractAndAppendId(res);
 };
 
 /**
  * A Datastore wrapper that gets all entities of a specified Kind.
  * @param kind The Kind that is being queried
+ * @param filters Any filters that will be applied to the query
  */
-const getAllWithId = async (kind: string) => {
-  const query = datastore.createQuery(kind);
+const getAllWithId = async (kind: string, filters?: Filter[]) => {
+  let query = datastore.createQuery(kind);
+  filters?.forEach(filter => {
```

Comment:
remove {}

Reply:
@ifhuang Oh hmmm I don't think I should do that because I'm just modifying query variable and not returning it!